### DISCUSSION TITLE
nosuid, noexec, nodev for /tmp

### DISCUSSION THREAD

**Discussion by m4rc2a** [Plebian-Linux/plebian-website#8] [Q&A]
How can I remount /tmp with special options? On my quartz model a is not /tmp not mounted extra so does this have any reasons? I only ask because do not want to break my system

**Comment by CounterPillow** [comment]
No specific reasons from Plebian's side. However, from Debian's side, [there is a concern](https://www.debian.org/doc/manuals/securing-debian-manual/ch04s10.en.html):

> Be careful if setting /tmp noexec when you want to install new software, since some programs might use it for installation. apt is one such program (see http://bugs.debian.org/116448) if not configured properly

So inferring from that, at least at the time when that manual was written, apt needed to be configured first to allow for a noexec /tmp.

The way to change the mount options in any case is to [just add the filesystem to fstab the normal way](https://www.freedesktop.org/wiki/Software/systemd/APIFileSystems/):

> Even though normally none of these API file systems are listed in `/etc/fstab` they may be added there. If so, any options specified therein will be applied to that specific API file system. Hence: to alter the mount options or other parameters of these file systems, simply add them to `/etc/fstab` with the appropriate settings and you are done.

By default, these special file systems aren't in fstab because systemd will take care of them during early boot-up, so specifying them in fstab would be redundant. If you want to change their mount options, you can do so by adding them to fstab however.
